content/concepts/consensus-network/invariant-checking.md

```diff
@@ -0,0 +1,404 @@
+# Invariant Checking
+
+This article provides a high-level overview of Invariant Checking, why it exist, how it works, and lists active invariant checks.
+
+When building applications on the XRP Ledger, it is crucial to understand the idea of Invariant Checks so as not to be surprised by specific error codes.
+
+## Introduction
+
+Invariant checking is a protection enhancement to the XRP Ledger and reinforces the critical properties of the XRP Ledger.
+
+Invariant checks should not trigger, but they ensure the XRP Ledger's integrity from bugs yet to be discovered or even created.
+
+
+## Why it Exists
+
+Nevertheless, why do we need invariant checks?
+
+- The overarching code for the XRP Ledger is complicated and vast; therefore, there is a high potential for code to execute incorrectly.
+- The cost of incorrectly executing a transaction is high and not acceptable by any standards.
+
+
+## How it Works
+
+A second layer of code runs automatically and in real-time after each transaction completes. It then examines the changes it made for correctness before the results are committed to the ledger. After every transaction runs, the invariant checker runs. Problematic transactions are marked with a **tecINVARIANT_FAILED** result code and are included in the ledger as having done nothing.
+
+The invariant checker is then invariantly checked to ensure the invariant checker's failure is not due to a bug. If this second level check fails, transactions are marked with a **tefINVARIANT_FAILED** result code. The transaction is in its final state and is not included in the ledger.
+
+
+## Active Invariant Checks
+
+Following is a list of active checks that the invariant checker runs against each transaction on the XRP Ledger.
```

Comment:
As a style thing, I suggest using "The following is..." instead of "Following is..." and ending with a colon instead of a period here. But we can also simplify this sentence more:

```suggestion
The XRP Ledger checks all the following invariants on each transaction:
```